pkg/operator/operator_config.go

```diff
@@ -335,15 +336,43 @@ func (r *operatorConfigReconciler) makeRuleEvaluatorConfig(ctx context.Context,
 		secretData[p] = b
 	}
 
-	cfg := &promconfig.Config{
-		GlobalConfig: promconfig.GlobalConfig{
-			ExternalLabels: labels.FromMap(spec.ExternalLabels),
+	// If no explicit project ID is set, use the one provided to the operator.
+	// On GKE the rule-evaluator can also auto-detect the cluster's project
+	// but this won't work in other Kubernetes environments.
+	queryProjectID, _, _ := resolveLabels(r.opts.ProjectID, r.opts.Location, r.opts.Cluster, spec.ExternalLabels)
+	if spec.QueryProjectID != "" {
+		queryProjectID = spec.QueryProjectID
+	}
+
+	var credentialsFile string
+	if spec.Credentials != nil {
+		credentialsFile = path.Join(secretsDir, pathForSelector(r.opts.PublicNamespace, &monitoringv1.SecretOrConfigMap{Secret: spec.Credentials}))
+	}
+
+	cfg := RuleEvaluatorConfig{
+		Config: promconfig.Config{
+			GlobalConfig: promconfig.GlobalConfig{
+				ExternalLabels: labels.FromMap(spec.ExternalLabels),
+			},
+			AlertingConfig: promconfig.AlertingConfig{
+				AlertmanagerConfigs: amConfigs,
+			},
+			RuleFiles: []string{path.Join(rulesDir, "*.yaml")},
 		},
-		AlertingConfig: promconfig.AlertingConfig{
-			AlertmanagerConfigs: amConfigs,
+		GoogleCloud: GoogleCloudConfig{
+			Query: &GoogleCloudQueryConfig{
+				ProjectID:       queryProjectID,
+				GeneratorURL:    spec.GeneratorURL,
+				CredentialsFile: credentialsFile,
+			},
 		},
-		RuleFiles: []string{path.Join(rulesDir, "*.yaml")},
 	}
+	if credentialsFile != "" {
+		cfg.GoogleCloud.Export = &GoogleCloudExportConfig{
+			CredentialsFile: ptr.To(credentialsFile),
+		}
+	}
```

Comment:
Want to kill `credentialsFile` and combine the 348 line here, for readability?

```suggestion
	if spec.Credentials != nil {
		cfg.GoogleCloud.Query.credentialsFile = path.Join(secretsDir, pathForSelector(r.opts.PublicNamespace, &monitoringv1.SecretOrConfigMap{Secret: spec.Credentials}))
		cfg.GoogleCloud.Export = &GoogleCloudExportConfig{
			CredentialsFile: ptr.To(credentialsFile),
		}
	}
```